### ISSUE TITLE
rename creates a stack overflow.

### ISSUE THREAD

**Issue by lloydtabb** [looker-open-source/malloy#195]
```
explore: words_base is table('malloy-data.wordle.words'){
  where: length(word) = 5 and  word ~ r'^[a-z]....$'

  join: numbers is table('malloy-data.malloytest.numbers'){dimension: a is 'a'; primary_key: a} on a
  dimension: a is 'a'

  rename: lower_word is word
  dimension: word is UPPER(word)

  query: words_and_position is {
    group_by: word is UPPER(word)
    nest: letters is {
      where: substr(word, numbers.num, 1) != ''
      order_by: 2
      group_by: [
        letter is substr(word, numbers.num, 1)
        position is numbers.num

      ]
    }
  }
}
```

**Comment by lloydtabb** [comment]
Actually there is an error in the program.  The stack overflow is probably much more simple to reproduce.

**Comment by mtoy-googly-moogly** [comment]
I can't reproduce this error. Please give detailed instructions on how to reproduce.

**Comment by mtoy-googly-moogly** [comment]
Actually not fixed, have the small test now. On it.

**Comment by mtoy-googly-moogly** [comment]
fixed with #221
